iceoryx_hoofs/include/iceoryx_hoofs/design_pattern/polymorphic_handler.hpp

```diff
@@ -39,7 +38,8 @@ struct DefaultHooks
     /// @brief called if the polymorphic handler is set or reset after finalize
     /// @param currentInstance the current instance of the handler singleton
     /// @param newInstance the instance of the handler singleton to be set
-    static void onSetAfterFinalize(Interface& currentInstance, Interface& newInstance) noexcept;
+    /// @note calls terminate and does not return
+    [[noreturn]] static void onSetAfterFinalize(Interface& currentInstance, Interface& newInstance) noexcept;
```

Comment:
I guess this does not restrict other hooks to define a non-noreturn method?

Reply:
No it does not and should not. It is not mandatory for any hook. It is just plugged in as a template argument type ...